tests/unit_tests/test_optical_thickness.py

```diff
@@ -0,0 +1,140 @@
+import openmc
+from openmc import RegularMesh
+import openmc.lib
+import numpy as np
+import openmc.mgxs as mgxs
+import pytest
+
+@pytest.fixture
+def my_model():
+    groups = mgxs.EnergyGroups(group_edges=[1e-5, 1.0e6])
+
+    scatter_matrix = np.array([[[0]]])
+
+    mat1_xsdata = openmc.XSdata('mat1', groups)
+    mat1_xsdata.order = 0
+    mat1_xsdata.set_total([0.01])
+    mat1_xsdata.set_absorption([0])
+    mat1_xsdata.set_scatter_matrix(scatter_matrix)
+
+    mat2_xsdata = openmc.XSdata('mat2', groups)
+    mat2_xsdata.order = 0
+    mat2_xsdata.set_total([0.02])
+    mat2_xsdata.set_absorption([0])
+    mat2_xsdata.set_scatter_matrix(scatter_matrix)
+
+    one_g_XS_file = openmc.MGXSLibrary(groups)
+    one_g_XS_file.add_xsdatas([mat1_xsdata, mat2_xsdata])
+    one_g_XS_file.export_to_hdf5('xs.h5')
+
+
+    # Create a dummy material with fictitious cross-section data
+    material1 = openmc.Material(name="Material1")
+    material1.add_macroscopic('mat1')
+
+    material2 = openmc.Material(name="Material2")
+    material2.add_macroscopic('mat2')
+
+    materials = openmc.Materials([material1, material2])
+    materials.cross_sections = 'xs.h5'
+    materials.export_to_xml()
+
+    sphere_inner = openmc.Sphere(r=2, boundary_type='transmission')
+    sphere_outer = openmc.Sphere(r=10, boundary_type='vacuum')
+
+    # Inner cell (filled with Material1)
+    cell_inner = openmc.Cell(name="Inner Sphere", fill=material1, region=-sphere_inner)
+
+    # Outer shell (filled with Material2)
+    cell_outer = openmc.Cell(name="Outer Sphere", fill=material2, region=+sphere_inner & -sphere_outer)
+
+    geometry = openmc.Geometry([cell_inner, cell_outer])
+    geometry.export_to_xml()
+
+
+    settings = openmc.Settings()
+    settings.energy_mode = "multi-group"
+    settings.particles = 1
+    settings.batches = 1
+    settings.export_to_xml()
+
+@pytest.fixture
+def lib_init(my_model):
+    openmc.lib.init()
+    yield
+    openmc.lib.finalize()
+
+@pytest.fixture
+def lib_simulation_init(lib_init):
+    openmc.lib.simulation_init()
+    yield
+
+def test_optical_thickness_variations(lib_init):
+    """Test optical thickness calculation for various scenarios."""
+    mat1_xs = 0.01
+    mat2_xs = 0.02
+    mat1_r = 2
+    mat2_r = 10
```

Comment:
Can these be defined in a test fixture and used where necessary throughout? That will allow you to try different values consistently